### PR TITLE
feat: allow passing the service-provider to the connection builder function

### PR DESCRIPTION
**Pull request by esskar**
**What this PR does / why we need it**:

The PR allows to use the IServiceProvider when build the signalr-hub-connection 

**Special notes for your reviewer**:

None

**Does this PR introduce a user-facing change?**:

It gives the user another extension method

### CODE REVIEW

**Comment by esskar** [comment]
> @esskar please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @dotnet-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @dotnet-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @dotnet-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@dotnet-policy-service agree

**Review by sungam3r — APPROVED** [review]

**Comment by sungam3r** [comment]
@esskar Thanks. I'll merge after fixing api approval test - just run it locally, delete approver file and then rename received file to approved.

**Comment by sungam3r** [comment]
ping @esskar , I would like to include this PR for v7.

**Comment by sungam3r** [comment]
Superseded by #1965

**Pull request closed without merging**
